docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,540 @@
+      action: DENY 
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: !dev
+    default:
+      action: DENY
+```
+
+See [how to generate a full rule-based view with negations](#how-to-generate-a-full-rule-based-view-with-negations)
+
+3. For each "rule" that has "action: ALLOW" generate a single "principal" in v3.RBAC:
+
+```yaml
+action: ALLOW
+policies:
+  "backend": # name of the most specific MeshTrafficPermission policy for the DPP
+    permissions:
+      - any: true
+    principals: # each principal is matched with OR semantics
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+```
+
+Similarly generate a v3.RBAC using "shadow_action: ALLOW". 
+Generated RBACs are set in the network filter as "rules" and "shadow_rules" respectively.
+
+4. If MeshTrafficPermission is targeting proxy then NetworkRBAC is placed as a NetworkFilter on the inbound listener.
+If MeshTrafficPermission is targeting MeshHTTPRoute then NetworkRBAC is placed as a NetworkFilter on the route. 
+Even in the case of MeshHTTPRoute, global per-proxy NetworkRBAC is still will be set to provide secure access for other routes. 
+
+### MeshTrafficPermission and Kubernetes
+
+MeshTrafficPermission is going to be the first policy with a new policy matching. 
+There are a few things worth mentioning regarding this fact. 
+
+#### Mesh as label
+
+Instead of creating a CRD with a "mesh" field, we can use a label to store mesh:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  labels:
+    kuma.io/mesh: mesh-1
+```
+
+This allows filtering policies by mesh using kubectl. 
+If the label is unset it will be added automatically by the webhook to be "kuma.io/mesh: default".
+
+#### Scope is namespaced
+
+Today all policies are cluster-scoped, but new policies should be namespace scoped.
+MeshTrafficPermission is limited to have only "kuma-system" namespace, 
+and affects DPPs across all namespaces.
+In the future we'll add support for other namespaces, so the policy could affect DPPs only in the specified namespace. 
+
+### Cross-mesh
+
+MeshTrafficPermission is used to specify a list of clients from other meshes that have an access to MeshGatewayRoute.
+In that case we use targetRef with "mesh" field:
+
+```yaml
+type: MeshTrafficPermission
+mesh: expose
+name: cross-mesh-permission
+spec:
+  targetRef:
+    kind: MeshGatewayRoute
+    name: cross-mesh-route
+  from:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        mesh: consume
+      default:
+        action: ALLOW
+    - targetRef:
+        kind: MeshServiceSubset
+        name: backend
+        mesh: consume
+        tags:
+          version: v1
+      default:
+        action: DENY
+```
```

Comment:
Feels like the mesh field should also be added in [005](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md). It should also be clear in the addition to 005 that having `mesh` unset means same mesh as the current policy and that setting a different mesh is dependent on the policy type and the main `targetRef`. 
These are details that are important to articulate as it's part of our decision!

Reply:
I added it yesterday, it was in the PR that changed `conf` to `default` https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md?plain=1#L66